command/agent/http.go

```diff
@@ -451,6 +451,9 @@ func errCodeFromHandler(err error) (int, string) {
 		} else if strings.HasSuffix(errMsg, structs.ErrTokenNotFound.Error()) {
 			errMsg = structs.ErrTokenNotFound.Error()
 			code = 403
+		} else if strings.HasSuffix(errMsg, structs.ErrJobRegistrationDisabled.Error()) {
+			errMsg = structs.ErrJobRegistrationDisabled.Error()
+			code = 403
```

Comment:
The intent of 409 is to avoid write skew and the direction in the standard is to adjust the request and submit it again. Which is exactly what we don't want here.

Reply:
I'm open to the idea of finding a more specific error code but I feel like needs to be one that doesn't tell any automated system "keep retrying!" The 403 tells them "_you_ can't submit because the permissions to submit have been changed"

Reply:
Yeah, I was afraid this might be the answer. I'll keep looking for a possible alternative code.